### PR TITLE
[BEAM-4019] Refactor HBaseIO splitting to produce ByteKeyRange objects

### PR DESCRIPTION
**Pull request by iemejia**
For some context this is an internal refactor towards a SDF based translation of HBase read, it is just internals. I extracted most (if not all) HBase region/split related methods into an individual class `HBaseUtils` too. I did not create an additional unit test for that one since it is well covered by HBaseIOTest and I did not want to add additional extra time to the build..

### CODE REVIEW

**Comment by iemejia** [comment]
R: @aromanenko-dev @tweise

File: sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseUtils.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.RegionLocator;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.util.Bytes;
+
+/** Utils to interact with an HBase cluster and get information on tables/regions. */
+class HBaseUtils {
+
```
[the remark this answers is not in the document]
Comment (iemejia) [reply sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseUtils.java]: Thanks, added docs in all methods of utils.

**Comment by iemejia** [comment]
Mmmm need to run some extra validations (eventual tests). This is apparently not yet good to merge.

File: sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java [hunk 2]
```diff
@@ -420,17 +313,38 @@ private long estimateSizeBytes() throws Exception {
...
+            HBaseUtils.getRegionLocations(connection, read.tableId, read.serializableScan.get());
+        LOG.debug("Suggested {} source(s) based on size", numSplits);
+        LOG.debug("Suggested {} source(s) based on number of regions", regionLocations.size());
+
+        List<ByteKeyRange> ranges =
+            HBaseUtils.getRanges(
+                regionLocations, read.tableId, read.serializableScan.get());
+        final int numSources = ranges.size();
+        LOG.debug("Spliting into {} source(s)", numSources);
+        if (numSources >= 1) {
```
Comment (tweise) [inline sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java]: `if numSources > 0` ?

Comment (iemejia) [reply sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java]: yes sir !

**Comment by iemejia** [comment]
Jenkins break is unrelated so I am merging it.

**Pull request merged**